src/components/layout/UserMenu.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @author sriram
+ *
+ * A component that renders user menu
+ *
+ *
+ */
+import React from "react";
+import { useTranslation } from "i18n";
+
+import ids from "./ids";
+import constants from "../../constants";
+
+import ExternalLink from "components/utils/ExternalLink";
+import { build } from "@cyverse-de/ui-lib";
+
+import {
+    Avatar,
+    Button,
+    Divider,
+    Grid,
+    Paper,
+    Typography,
+    useTheme,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme) => ({
+    paper: {
+        padding: theme.spacing(1),
+        [theme.breakpoints.down("xs")]: {
+            padding: theme.spacing(0.2),
+            backgroundColor: theme.palette.info.main,
+        },
+    },
+
+    typography: {
+        color: theme.palette.info.main,
+        [theme.breakpoints.down("xs")]: {
+            color: theme.palette.info.contrastText,
+        },
+    },
+
+    divider: {
+        margin: theme.spacing(1),
+        [theme.breakpoints.down("xs")]: {
+            margin: theme.spacing(0.5),
+        },
+    },
+
+    button: {
+        color: theme.palette.primary.main,
+        [theme.breakpoints.down("xs")]: {
+            color: theme.palette.info.contrastText,
+        },
+    },
+}));
+export default function UserMenu(props) {
+    const {
+        profile: {
+            id,
+            attributes: { email, name },
+        },
```

Comment:
Instead of having `userProfile` passed in as a prop, couldn't this component use the `useUserProfile` hook for itself?

Reply:
We can do that for sure. Only downside I can think of is that it will involve checking profile to see if  it null or not which is already happening in `CyverseAppBar` in several places. Otherwise I can't think of any reason either in favor or against.